enclave-runtime/src/test/evm_pallet_tests.rs

```diff
@@ -0,0 +1,238 @@
+
+	let trusted_call = TrustedCall::evm_create(
+		sender_acc.clone(),
+		sender_evm_acc,
+		Vec::from_hex(smart_contract.to_string()).unwrap(),
+		U256::from(0),
+		10_000_000,    // gas limit
+		U256::from(1), // max_fee_per_gas !>= min_gas_price defined in runtime
+		None,
+		Some(U256::from(0)),
+		Vec::new(),
+	)
+	.sign(&sender.clone().into(), 0, &mrenclave, &shard);
+
+	// when
+	Stf::execute(&mut state, trusted_call, &mut opaque_vec, [0u8, 1u8]).unwrap();
+
+	// then
+	let execution_address = H160::from_slice(
+		&Vec::from_hex("0xce2c9e7f9c10049996173b2ca2d9a6815a70e890".to_string()).unwrap(),
+	);
+
+	assert!(state.execute_with(|| get_evm_account_codes(&execution_address).is_some()));
+
+	let counter_value = state
+		.execute_with(|| get_evm_account_storages(&execution_address, &H256::zero()))
+		.unwrap();
+	assert_eq!(H256::from_low_u64_be(1), counter_value);
+	let last_caller = state
+		.execute_with(|| get_evm_account_storages(&execution_address, &H256::from_low_u64_be(1)))
+		.unwrap();
+	assert_eq!(H256::from(sender_evm_acc), last_caller);
+
+	// Call to inc() function
+	{
+		// in solidity compile information you get the hash of the call
+		let function_hash = Vec::from_hex("371303c0".to_string()).unwrap();
+
+		let inc_call = TrustedCall::evm_call(
+			sender_acc.clone(),
+			sender_evm_acc,
+			execution_address,
+			function_hash,
+			U256::from(0),
+			10_000_000,    // gas limit
+			U256::from(1), // max_fee_per_gas !>= min_gas_price defined in runtime
+			None,
+			Some(U256::from(1)),
+			Vec::new(),
+		)
+		.sign(&sender.clone().into(), 1, &mrenclave, &shard);
+		Stf::execute(&mut state, inc_call, &mut opaque_vec, [0u8, 1u8]).unwrap();
+
+		let counter_value = state
+			.execute_with(|| get_evm_account_storages(&execution_address, &H256::zero()))
+			.unwrap();
+		assert_eq!(counter_value, H256::from_low_u64_be(2));
```

Comment:
Can we extract a common function here? It seems that this part is duplicated a couple of times.